### ISSUE TITLE
Low loss, bad texts in Streaming Conformer ASR

### ISSUE THREAD

**Issue by Stasiche** [espnet/espnet#4070]
Hello! I'm trying to learn  ESPnetASRModel with [contextual_block_conformer_encoder](https://github.com/espnet/espnet/blob/master/espnet2/asr/encoder/contextual_block_conformer_encoder.py) by myself. I mean, I've wrote dataloader and train loop by myself.
The problem that I've got quite low loss (several dozen), but very bad quality of the ASR (couple of repeted tokens or blank line). Have you got any idea what is wrong?

I pad speech with zeros and texts with "-1"

**Comment by Stasiche** [comment]
I'm trying to overfit under LibriSpeech dev-clean and get this loss
![image](https://user-images.githubusercontent.com/26529547/153771847-e8d0bea3-cef9-4c97-aa22-5da024e72f7b.png)

**Comment by Stasiche** [comment]
But predicted text is (the first line -- prediction/ second -- reference text)
=========

null

YES

=========

HE SAID

HIS FRIENDS EYELIDS FLICKERED

=========

HE UP

ITS THAT AS MAKES EM I WONT SAY WHAT

=========

HE DOOR

SUCH PERFECT COMPLETENESS IS NOT IN NATURE

=========

HE HAD THEM

HE ISNT FIT TO HEAR WHATS SAID HERE HE BELIEVES ITS ALL TRUE

=========

MY

AND THE NEW BLOOD AT THE ERA OF THIS STORY WAS AT HAND

=========

WHAT WAS LORD

ROUND HIS BODY WAS ROLLED A LONG COIL OF FLAX ROPE

=========

GEORGE NODDED

HE WANTS ME TO STAND READY TO OBEY ANY SUMMONS THE POLICE MAY SEND ME

=========

HE INDEED SOMETHING ELSE

SHE BEGAN TO SHOW THEM HOW TO WEAVE THE BITS OF THINGS TOGETHER INTO NESTS AS THEY SHOULD BE MADE

=========

HE DRANK NORTH

HE PRESSED HIS VIOLIN CASE TO HIS HEART AS IF IT WERE A LIVING THING THAT COULD KNOW THAT HE LOVED IT

=========

I CANT DO IT

THROUGH THE INFLUENCE OF HON THOMAS L HAMER HE WAS ADMITTED AT WEST POINT IN EIGHTEEN THIRTY NINE

=========

WAS

LOOK A LITTLE CLOSER WHILE OUR GUIDE LETS THE LIGHT OF HIS LAMP FALL UPON THE BLACK WALL AT YOUR SIDE

=========

AND CANT SUCKLE A

WHEN A MARRIED WOMAN HAS FOLLOWERS AND THE HUSBAND DONT GO THE WRONG SIDE OF THE POST TOO OR IT AINT PROVED AGAIN HIM THAT HE DO THEYLL NEVER LET HER HAVE NOTHING TO DO WITH THE CHILDREN

=========

DRYS

TO THE PICKLE ADD TWO LARGE ONIONS CUT IN QUARTERS TWO FRESH CARROTS AND ABOUT ONE OUNCE OF MIXED WHOLE ALLSPICE BLACK PEPPERS CLOVES AND BAY LEAVES

=========

THE DRANKS

THE HIGHEST INSTINCT FOR PURITY PLACES HIM WHO IS AFFECTED WITH IT IN THE MOST EXTRAORDINARY AND DANGEROUS ISOLATION AS A SAINT FOR IT IS JUST HOLINESS THE HIGHEST SPIRITUALIZATION OF THE INSTINCT IN QUESTION

=========

**Comment by Stasiche** [comment]
the problem was that I was using encoder-decoder architecture. I should use rnn-t
